### PR TITLE
LPS-157177 | Object entry deletion impossible with prevent deletionType relationship created

### PR DESCRIPTION
**Pull request by Hijky**
**Background**
Add a test to assert object entry deletion impossible with prevent deletionType relationship created

**Test Plan**
Given university created
And Given subject created
And Given manyToMany relationship universitySubjects with deletionType prevent created
And Given three university entries created
And Given subject entry created
And Given subject related to all three universities
When I call DELETE to /o/c/subjects/{subjectId}
Then the response is 400 and the message is "Object relationship {subjectId}
does not allow deletes"

**JIRA Ticket**
https://issues.liferay.com/browse/LPS-157177

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:headless-frontend-functional:nocompile

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase [hunk 1]
```diff
@@ -794,4 +798,81 @@ definition {
...
+				relationshipName = "universitiesSubjects");
+		}
+
+		task ("When deleting subject entry") {
+			var responseToParse = ObjectDefinitionAPI.deleteObjectEntry(
+				en_US_plural_label = "subjects",
+				objectEntryId = "${subjectId}");
+		}
+
+		task ("Then assert bad response and subject entry is not deleted") {
```
Comment (john-co) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase:862]: Test plan also asserts for `and the message is "Object relationship {subjectId}
does not allow deletes"`. I do like the `assertResponseIncludeCorrectDetailsOfNotDeletedObject` to double check, but I don't see where the assertion for the particular message is made.

**Review by john-co — COMMENTED** [review]
Looks good overall to me. Seemed like there was a missing assertion at the end and some minor improvement suggestions.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase [hunk 2]
```diff
@@ -794,4 +798,81 @@ definition {
 		}
 	}
 
+	@priority = "4"
+	test ObjectEntryDeletionImpossibleWithPreventDeletionTypeRelationshipCreated {
+		property portal.acceptance = "true";
+
+		task ("Given manyToMany relationship universitySubjects created") {
+			var objectDefinitionId1 = JSONObject.getObjectId(objectName = "University");
+			var objectDefinitionId2 = JSONObject.getObjectId(objectName = "Subject");
```
[the remark this answers is not in the document]
Comment (Hijky) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase:807]: Sure, it seems a good idea, I will edit it.

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase (hunk 1 above)
Comment (Hijky) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase:862]: This problem is still not consistent with the test step in my manual test last week, I will continue to pay attention this week

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase (hunk 2 above)
[the remark this answers is not in the document]
Comment (Hijky) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase:807]: I test it, and it seems the var value in the setup can not pass into

[the remark this answers is not in the document]
Comment (john-co) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase:807]: oh, you can modify the variable in setup phase to be a global variable with `static var =` 

https://liferay.slack.com/archives/CD7939WBE/p1639501193201500?thread_ts=1639500966.200800&cid=CD7939WBE

[the remark this answers is not in the document]
Comment (Hijky) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase]: Thanks, I modify it successfully

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-frontend-functional:nocompile

File: portal-web/test/functional/com/liferay/portalweb/macros/headless/objects/ObjectDefinitionAPI.macro [hunk 3]
```diff
@@ -567,6 +621,25 @@ definition {
 		return "${response}";
 	}
 
+	macro setUpGlobalobjectDefinitionId {
```
Comment (magjed4289) [inline portal-web/test/functional/com/liferay/portalweb/macros/headless/objects/ObjectDefinitionAPI.macro:624]: really nice! 👏🏻

**Review by magjed4289 — APPROVED** [review]
looks good! thank you! :)

**Comment by Hijky** [comment]
ci:forward

**Pull request closed without merging**